rs/crypto/secp256k1/src/lib.rs

```diff
@@ -620,6 +620,20 @@ impl PrivateKey {
         Ok(tweaked_key.sign_message_with_bip340(message, rng))
     }
 
+    /// Sign a message with BIP340 Schnorr with Taproot derivation, without using an external RNG
+    ///
+    /// The aux_rand parameter for BIP340 is just to re-randomize the signature,
+    /// and may prevent certain forms of fault attack. It it is otherwise not necessary
+    /// for security.
```

Comment:
After reading this sentence, I'm not fully sure what it wants to say. The `aux_rand` comes a bit out of the blue. Also, it is not fully clear that the external RNG is used for the re-randomization of the signature (it is somewhat implicit though). And ideally we'd say what the exact implications are of not using the `no_rng` (or `fixed_rng`) variant, i.e., how bad these "certain forms of fault attacks" are.

If this is only sufficient for testing (such as for the state machine tests), then it might make sense to make that explicit in the name or at least in the Rustdoc.